src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -634,31 +741,50 @@ private static VCFHeader warpHeader(VCFHeader in, Map<String, Long> namesAndLeng
    * then performs liftover on BED, then classifies regions
    * based on pre- and post- liftover regions.
    */
-  private static List<HomologousRange> processAndSaveBed(String inputBed, Fasta queryFasta,
+  private static List<HomologousRange> processAndSaveBed(String inputBed, String inputVcf,
+      Fasta queryFasta,
       Fasta targetFasta) {
     // Open necessary readers
     BufferedReader bedReader = null;
+    VCFFileReader vcfReader = null;
     try {
       logger.log(Level.INFO, "Reading BED");
       bedReader = Files.newBufferedReader(Paths.get(inputBed), UTF_8);
+      logger.log(Level.INFO, "Reading VCF");
+      vcfReader = new VCFFileReader(new File(inputVcf), false);
     } catch (IOException ex) {
       fail("failed to parse input BED/FASTA/VCF file(s): " + ex.getMessage());
     }
 
     // Takes the input BED and splits at non-DNA characters
     logger.log(Level.INFO, "Split DNA at non-DNA characters");
-    List<GenomeRange> dnaOnlyBED = null;
+    List<GenomeRange> dnaOnlyInputBED = null;
     try {
-      if ((dnaOnlyBED = splitAtNonDNA(queryFasta, bedReader)) == null) {
+      if ((dnaOnlyInputBED = splitAtNonDNA(queryFasta, bedReader)) == null) {
         fail("failed to generate reference genome");
       }
     } catch (IOException ex) {
       fail("failed to read from input bed: " + ex.getMessage());
     }
 
+    logger.log(Level.INFO, "Generating regions from variants");
+    List<GenomeRange> fromVcfBED = generateBEDFromVCF(vcfReader);
+
+    Collections.sort(fromVcfBED);
+    Collections.sort(dnaOnlyInputBED);
+
+    logger.log(Level.INFO, "Removing overlap from VCF ranges");
+    fromVcfBED = omitOverlap(fromVcfBED);
```

Comment:
We don't want to omitOverlap here -- that throws away all records that have overlap with each other. Instead, here we want to merge overlapping records. E.g. input of

chr1   10   20
chr1   15   25
chr2   20   30

should produce output of two records:
chr1   10   25
chr2   20   30